packages/core/src/dynamoDb/utils/editItem.ts

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs'
+import * as os from 'os'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { RowData, TableSchema, getAttributeValue } from './dynamodb'
+import { DynamoDbClient } from '../../shared/clients/dynamoDbClient'
+import { AttributeUpdates, GetItemInput, GetItemOutput, UpdateItemInput } from 'aws-sdk/clients/dynamodb'
+import { telemetry } from '../../shared/telemetry'
+
+export async function editItem(
+    selectedRow: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema }
+) {
+    const currentContent = JSON.stringify(selectedRow, undefined, 4)
+    let filename = `${selectedRow[tableSchema.partitionKey.name]}`
+    if (tableSchema.sortKey) {
+        filename += `-${selectedRow[tableSchema.sortKey.name]}`
+    }
+    const tempFilePath = path.join(os.tmpdir(), `${filename}.json`)
+    fs.writeFileSync(tempFilePath, currentContent)
+
+    const document = await vscode.workspace.openTextDocument(tempFilePath)
+    await vscode.window.showTextDocument(document, { preview: false, viewColumn: vscode.ViewColumn.Beside })
+
+    const saveListener = vscode.workspace.onDidSaveTextDocument(async (savedDoc) => {
+        if (savedDoc.fileName === tempFilePath) {
+            await saveItem(tempFilePath, selectedRow, { tableName, regionCode, tableSchema })
+        }
+    })
+
+    const closeListener = vscode.workspace.onDidCloseTextDocument(async (closedDoc) => {
+        if (closedDoc.fileName === tempFilePath) {
+            await saveItem(tempFilePath, selectedRow, { tableName, regionCode, tableSchema })
+            saveListener.dispose()
+            closeListener.dispose()
+            if (fs.existsSync(tempFilePath)) {
+                fs.unlinkSync(tempFilePath)
+            }
+        }
+    })
+}
+
+async function saveItem(
+    tempFilePath: string,
+    selectedRow: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema }
+) {
+    try {
+        const updatedContent = JSON.parse(fs.readFileSync(tempFilePath, 'utf-8'))
+        const response = await updateDynamoDbItem(updatedContent, selectedRow, { tableName, regionCode, tableSchema })
+        if (response) {
```

Comment:
We probably don't need to notify users if the item was successfully updated because thats already their expectation + we can see it updated in the ui